### PR TITLE
ET-3888 add alarms

### PR DESCRIPTION
**Pull request by Robert-Steiner**
What:

- fix budget notification
- adds basic notifications for:

API Gateway:
- Average IntegrationLatency > 250ms within 1min
- Average Latency > 300ms within 1min
- 5XXErrors > 0 within 1min

ECS Services:
- front office CPU usage > 90% within 3min
- back office CPU usage > 90% within 3min
- front office log errors > 0  within 1min
- back office log errors > 0  within 1min

ALB
- HTTPCode_Target_ELB_5XX_Count > 0 within 1min (the service returns 5xx)
- HTTPCode_ELB_4XX_Count > 0 within 1min (the alb returns 4xx)
- HTTPCode_ELB_5XX_Count > 0 within 1min (the alb returns 5xx) 

WAF
- BlockedRequests for ip-rate-limit > 0 (ip hit rate limit)
- CountedRequests for ALL request > 40k within 1min
- BlockedRequests for ALL request > 5k within 1min

Postgres
- Average ReadLatency > 10ms within 1min
- Average WriteLatency > 10ms within 1min

AWS Budget
-  actual cost > 80% of budget limit
-  forecasted cost > 100% of budget limit

### CODE REVIEW

File: b2b/alarms/global/variables.tf [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+variable "create_waf_alarms" {
+  description = "Whether to create WAF alarms"
+  type        = bool
+  default     = true
+}
+
+variable "web_acl" {
+  description = "Name of the WebALC"
```
Comment (jorge-cr-13) [inline b2b/alarms/global/variables.tf:43]:
```suggestion
  description = "Name of the WebACL"
```

File: b2b/api_gateway/proxy/main.tf [hunk 2]
```diff
@@ -207,3 +211,88 @@ resource "aws_wafv2_web_acl_association" "api_gateway" {
...
+# cloudwatch alarms
+module "http_5xx_error_alarm" {
+  count   = var.create_alarms ? 1 : 0
+  source  = "terraform-aws-modules/cloudwatch/aws//modules/metric-alarm"
+  version = "4.2.1"
+
+  alarm_name          = "api_gateway_5xx_error"
+  alarm_description   = "Number of API Gateway HTTP-5XX errors > ${var.http_5xx_error_threshold}. It may indicate an issue within the NLB integration or with the lambda authorizer."
+  comparison_operator = "GreaterThanThreshold"
+  evaluation_periods  = 1
```
Comment (joecks) [inline b2b/api_gateway/proxy/main.tf:224]: everywhere else you used 2, but I guess this relates to the statistic field that is SUM here. But would that mean that we only create the sum over 1 period?

**Review by joecks — APPROVED** [review]
Looks good to me

File: b2b/api_gateway/proxy/main.tf (hunk 2 above)
Comment (Robert-Steiner) [reply b2b/api_gateway/proxy/main.tf:224]: It means if the SUM of errors in one period is over the threshold the alarm is triggered. https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/AlarmThatSendsEmail.html#alarm-evaluation

**Review by joecks — APPROVED** [review]

**Pull request merged**